bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go

```diff
@@ -277,10 +284,11 @@ func (r *KubeadmConfigReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 				return r.rotateMachinePoolBootstrapToken(ctx, config, cluster, scope)
 			}
 		}
-		// In any other case just return as the config is already generated and need not be generated again.
-		return ctrl.Result{}, nil
+		// If config.Status.Ready, just return as the config is already generated and need not be generated again.
+		if config.Status.Ready {
+			return ctrl.Result{}, nil
+		}
```

Comment:
This probably makes sense, although doesn't fully remove the dependency on the status.Ready field, like the title of the PR says, although that's probably ok, given that the code from here onwards just goes all the way to create the Secret, and ultimately update status if the secret isn't there.

Reply:
+1 , I have updated the title to reflect code change . Thanks.